userspace/libscap/examples/01-open/test.c

```diff
@@ -336,6 +350,118 @@ void print_parameter(int16_t num_param)
+	for(int i = 0; i < dim; i++)
+	{
+		printf("[%d] %s\n", i, string_vector[i]);
+	}
+	printf("Interesting syscalls: %d\n", dim);
+}
+
+/* This are the real interesting syscalls that we want to support in the new probe.
+ * - all syscalls associated with events of type `UF_NEVER_DROP`.
+ * - all syscalls that are not managed through `GENERIC_EVENTS` and don't
+ *   have the `EF_DROP_SIMPLE_CONS` flag.
+ *
+ * Please note: if some syscalls miss probably you have an old kernel
+ * that don't define them. Try to use a newer one.
+ */
+
+void print_modern_probe_syscalls()
+{
+	char str[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_SIZE];
+	int interesting_syscall = 0;
+	enum ppm_syscall_code ppm_syscall_code = 0;
+
+	/* For every syscall of the system. */
+	for(int syscall_id = 0; syscall_id < SYSCALL_TABLE_SIZE; syscall_id++)
+	{
+		/* TAKE ALWAYS: If the syscall has `UF_NEVER_DROP` flag we cannot use simple consumer. */
+		if(g_syscall_table[syscall_id].flags & UF_NEVER_DROP)
+		{
+			ppm_syscall_code = g_syscall_code_routing_table[syscall_id];
+			if(!g_syscall_info_table[ppm_syscall_code].name)
+			{
+				printf("*ERROR: a `UF_NEVER_DROP` syscall has no the corresponding name in the g_syscall_info_table.\n");
+				exit(EXIT_FAILURE);
+			}
+			strcpy(str[interesting_syscall++], g_syscall_info_table[ppm_syscall_code].name);
+			continue;
+		}
+
+		/* TAKE NEVER: If we use generic events, we can drop the syscall with the simple consumer logic. */
+		if(g_syscall_table[syscall_id].enter_event_type == PPME_GENERIC_E)
+		{
+			continue;
+		}
+
+		ppm_syscall_code = g_syscall_code_routing_table[syscall_id];
+		if(g_syscall_info_table[ppm_syscall_code].flags & EF_DROP_SIMPLE_CONS)
+		{
+			continue;
+		}
+
+		/* This is an error since it means that a syscall we want to trace is not tracked in our `g_syscall_info_table`.
+		 * We have `EC_UNKNOWN` when we don't have an entry in the `g_syscall_info_table`.
+		 */
+		if(g_syscall_info_table[ppm_syscall_code].category == EC_UNKNOWN)
+		{
+			printf("*ERROR: a syscall that has an event associated is unknown in g_syscall_info_table.\n");
+			exit(EXIT_FAILURE);
+		}
+
+		strcpy(str[interesting_syscall++], g_syscall_info_table[ppm_syscall_code].name);
+	}
+
+	print_ordered_syscalls(str, interesting_syscall);
+}
+
+/* Print all supported syscall. Please if you only want
+ * to see syscalls in simple consumer mode, enable the
+ * `--simple_consumer` option.
+ */
+void print_kernel_simple_consumer_syscalls()
+{
+	bool interesting[PPM_SC_MAX];
+	char str[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_SIZE];
+	int interesting_syscall = 0;
+
+	for(int i = 0; i < PPM_SC_MAX; i++)
+	{
+		for(int syscall_nr = 0; syscall_nr < SYSCALL_TABLE_SIZE; syscall_nr++)
+		{
+			if(g_syscall_code_routing_table[syscall_nr] == i)
+			{
+				if(args.ppm_sc_of_interest.ppm_sc[i] || g_syscall_table[syscall_nr].flags & UF_NEVER_DROP)
+				{
+					strcpy(str[interesting_syscall++], g_syscall_info_table[i].name);
+				}
+			}
```

Comment:
This is an extreme nitpick, but I think either combining these if conditions or using a continue on the first one might help readability by reducing the number of nested statements.

Reply:
That's true, I will surely do it, thank you